src/background/controller/wallet.ts

```diff
@@ -2218,33 +2215,52 @@ export class WalletController extends BaseController {
       fcl.arg(gasLimit, t.UInt64),
     ]);
 
+    let evmAddress = await this.getEvmAddress();
+
     mixpanelTrack.track('ft_transfer', {
-      from_address: await this.getEvmAddress(),
+      from_address: evmAddress,
       to_address: to,
       amount: parseFloat(amount),
       ft_identifier: 'FLOW',
       type: 'evm',
     });
 
-    console.log('result ', result);
-    const res = await fcl.tx(result).onceSealed();
-    const transactionExecutedEvent = res.events.find((event) =>
-      event.type.includes('TransactionExecuted')
-    );
+    if (evmAddress.startsWith('0x')) {
+      evmAddress = evmAddress.substring(2);
+    }
+
+    const addressNonce = await this.getNonce(evmAddress);
 
-    if (transactionExecutedEvent) {
-      const hash = transactionExecutedEvent.data.hash;
-      const hashHexString = hash.map((num) => parseInt(num).toString(16).padStart(2, '0')).join('');
+    const keccak256 = (data: Buffer) => {
+      return ethUtil.keccak256(data);
+    };
 
+    // [nonce, gasPrice, gasLimit, to.addressData, value, data, v, r, s]
+    const directCallTxType = 255;
+    const contractCallSubType = 5;
+    const noceNumber = Number(addressNonce);
+    const gasPrice = 0;
+    const transactionValue = BigInt(amount * 10 ** 18);
```

Comment:
Above we already convert hex to number, we can use it redirectly.
the value*10**18 is the example code, I don't wanna add web3js lib.
```js
   // Convert the hex value to number
      const number = web3.utils.hexToNumber(value);
```